### DISCUSSION TITLE
Issues with State Triggers

### DISCUSSION THREAD

**Discussion by clbx** [custom-components/pyscript#659] [Q&A]
I'm having issues getting anything to trigger with state triggers and I'm unsure why 

I have a simple script

```
@state_trigger("sensor.beauty_room_bathroom_switch_action")
def small_bathroom_switch():
    log.info("HIT")
```

I've tried the state trigger as is and with 
```
@state_trigger("sensor.beauty_room_bathroom_switch_action == '1_short_release')
```

But I can never get anything to trigger,

It appears that pyscript is loading my script

```
2024-11-18 01:13:06.048 DEBUG (MainThread) [custom_components.pyscript.trigger] trigger scripts.buttons.small_bathroom.small_bathroom_switch: watching vars {'sensor.beauty_room_bathroom_switch_action'}
2024-11-18 01:13:06.048 DEBUG (MainThread) [custom_components.pyscript.trigger] trigger scripts.buttons.small_bathroom.small_bathroom_switch waiting for state change or event
```

And Home Assistant is registering that a change is made 
<img width="344" alt="image" src="https://github.com/user-attachments/assets/ec0c0587-7049-4517-b7bd-903525b51fec">

and I have the correct entity ID
<img width="576" alt="image" src="https://github.com/user-attachments/assets/81538391-2a0a-4661-8557-f0d9afb5a91d">

What am I doing wrong??

**Comment by ALERTua** [comment]
try an isolated example: create a helper entity of toggle type and name it `test_boolean`. the newly created entity `input_boolean.test_boolean` will appear. then try to `state_trigger` on this boolean with
```
@state_trigger("input_boolean.test_boolean")
def test_boolean_trigger():
    log.warning("TEST BOOLEAN HIT")
```
pasted to an empty file `test_boolean.py` at `/config/pyscript/`
save the file, restart your home assistant to be 100% sure.
then change the boolean value and look at the logs.
will the message appear?

**Reply by clbx** [reply]
I do not see the message

**Reply by ALERTua** [reply]
I'd suggest double-checking the installation procedure.

**Comment by Wackoww** [comment]
Maybe you need to specify if and how pyscript will use logging.
See documentation https://hacs-pyscript.readthedocs.io/en/stable/reference.html#logging
